pkg/distributor/distributor.go

```diff
@@ -78,6 +78,11 @@ var (
 	})
 )
 
+type clusterTracker interface {
+	LookupReplica(ctx context.Context, cluster string, replica string) *replicaTracker
```

Comment:
This need to return an error.  You never swallow errors.

Reply:
Lookup implies this is side effect free.  Probably want to go with `CheckReplica` or something?

Reply:
And probably not return the replica, as its checking it validity.  Just return an error?